packages/main/src/plugin/image-layers.ts

```diff
@@ -0,0 +1,48 @@
+import nodeTar from 'tar';
+import fs from 'node:fs';
+import path from 'node:path';
+import { fileTree } from './filetree.js';
+
+export interface ImageLayer {
+  id: string;
+  history?: string;
+  files: nodeTar.ReadEntry[];
+  tree: fileTree<nodeTar.ReadEntry>;
+}
+
+export async function getLayersFromImageArchive(tmpdir: string): Promise<ImageLayer[]> {
```

Comment:
I think it would be easier with a class for testing (as it's missing unit tests there)